controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -2988,5 +2989,52 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
         };
       });
     },
+
+    isGitHubAppInstalled: (req, ctx) => {
+      const logger = opts.logger.child({
+        service: ctx.service.typeName,
+        method: ctx.method.name,
+      });
+
+      return handleError<PlainMessage<IsGitHubAppInstalledResponse>>(logger, async () => {
+        const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
+        const orgRepository = new OrganizationRepository(opts.db);
+
+        if (!opts.githubApp) {
+          return {
+            response: {
+              code: EnumStatusCode.ERR,
+              details: 'GitHub app integration is disabled',
+            },
+            isInstalled: false,
+          };
+        }
+
+        const org = orgRepository.byId(authContext.organizationId);
+        if (!org) {
+          return {
+            response: {
+              code: EnumStatusCode.ERR_NOT_FOUND,
+              details: 'Organization not found',
+            },
+            isInstalled: false,
+          };
+        }
+
+        const githubRepository = new GitHubRepository(opts.db, opts.githubApp);
+        const isInstalled = await githubRepository.isAppInstalledOnRepo({
+          accountId: req.gitInfo?.accountId ?? '',
```

Comment:
How can it be empty?

Reply:
It can never be, atleast in this context. protobuf generates optional fields sadly. Added a check just in case.

Reply:
ok